### DISCUSSION TITLE
Trying to add a label

### DISCUSSION THREAD

**Discussion by Hildepont-0** [LeMyst/WikibaseIntegrator#92] [Q&A] [answered]
Hello,
I'm trying to create a new item and add a specific label :

> a = wbi_core.String(value='test 2', prop_nr='P23')
> b = [a]
> new_item = wbi_core.ItemEngine(data=b)
> new_item.set_label('new_label','en','REPLACE')

unfortunately, I get the following message : 

> 
> Traceback (most recent call last):
>   File "test.py", line 37, in <module>
>     new_item.set_label('new_label','en','REPLACE')
>   File "/usr/local/lib/python3.8/dist-packages/wikibaseintegrator-0.10.0.dev0-py3.8.egg/wikibaseintegrator/wbi_core.py", line 400, in set_label
> AttributeError: 'NoneType' object has no attribute 'get_language_data'

Am I guilty ?

**Comment by LeMyst** [comment, marked as answer]
Hi @Hildepont-0 ,

I found the issue. I never encountered it because I always use the "Fast Run" mode in my scripts.
I'm trying to add some unit tests to correctly fix that part.
Expect the fix today.

Thanks for your report

**Reply by LeMyst** [reply]
Issue #94 and PR #93 created

**Reply by LeMyst** [reply]
Fix pushed

**Reply by Hildepont-0** [reply]
Thank you for the support ! I think that the set_label function is now ok since the following lines work :

> a = wbi_core.String(value='James Ensor', prop_nr='P24')
> data = [a]
> wd_item = wbi_core.ItemEngine(data=data)
> wd_item.set_label('James Ensor','en','REPLACE')
> print(type (wd_item))

But when I try to write to the wikibase nothing works anymore (even without set_label) : 

> login_instance = wbi_login.Login(user='WikibaseAdmin@bot_1', pwd='psswd')
> a = wbi_core.String(value='James Ensor', prop_nr='P24')
> b = [a]
> new_item = wbi_core.ItemEngine(data=b)
> new_item.write(login_instance)

It worked before. Now it gives me :

> /usr/local/lib/python3.8/dist-packages/wikibaseintegrator-0.10.0.dev0-py3.8.egg/wikibaseintegrator/wbi_core.py:1393: UserWarning: Warning: No distinct value properties found
> Please set P2302 and Q21502410 in your Wikibase or set `core_props` manually.
> Continuing with no core_props
> {'action': 'wbeditentity', 'data': '{"claims": {"P24": [{"mainsnak": {"snaktype": "value", "property": "P24", "datavalue": {"value": "James Ensor", "type": "string"}, "datatype": "string"}, "type": "statement", "rank": "normal", "qualifiers": {}, "qualifiers-order": [], "references": []}]}}', 'format': 'json', 'token': 'd307f1ad2077d3c71a117b8502731d666013e138+\\', 'summary': '', 'maxlag': 5, 'bot': '', 'new': 'item'}
> Error while writing to the Wikibase instance
> Traceback (most recent call last):
>   File "test.py", line 40, in <module>
>     new_item.write(login_instance)
>   File "/usr/local/lib/python3.8/dist-packages/wikibaseintegrator-0.10.0.dev0-py3.8.egg/wikibaseintegrator/wbi_core.py", line 668, in write
>   File "/usr/local/lib/python3.8/dist-packages/wikibaseintegrator-0.10.0.dev0-py3.8.egg/wikibaseintegrator/wbi_core.py", line 1177, in mediawiki_api_call_helper
>   File "/usr/local/lib/python3.8/dist-packages/wikibaseintegrator-0.10.0.dev0-py3.8.egg/wikibaseintegrator/wbi_core.py", line 1040, in mediawiki_api_call
> wikibaseintegrator.wbi_core.MWApiError: {'error': {'code': 'mustpostparams', 'info': 'The following parameter was found in the query string, but must be in the POST body: token.', '*': 'See http://localhost:8181/w/api.php for API usage. Subscribe to the mediawiki-api-announce mailing list at &lt;https://lists.wikimedia.org/mailman/listinfo/mediawiki-api-announce&gt; for notice of API deprecations and breaking changes.'}}

**Reply by Hildepont-0** [reply]
I confirm this error doesn't appear with the 'old' code (25 January)

**Reply by Hildepont-0** [reply]
also, same problem when I log in as a normal user (user='WikibaseAdmin') instead of a bot

**Reply by LeMyst** [reply]
Hello @Hildepont-0 
Sorry for the late answer, I missed your message.
I pushed a fix that will solve your issue

Thanks four your report

**Reply by Hildepont-0** [reply]
@LeMyst 
thank you for your edits, it works now like a charm !
